docs/developing-locally.rst

```diff
@@ -9,24 +9,53 @@ Setting Up Development Environment
 
 Make sure to have the following on your host:
 
-* virtualenv_;
-* pip;
-* PostgreSQL.
+* Python 3.6
+* PostgreSQL_.
+* Redis_, if using Celery
 
 First things first.
 
-#. `Create a virtualenv`_.
+#. Create a virtualenv: ::
 
-#. Activate the virtualenv you have just created.
+    $ python3 -m venv <virtual env path>
+
+#. Activate the virtualenv you have just created: ::
+
+    $ source <virtual env path>/bin/activate
 
 #. Install development requirements: ::
 
     $ pip install -r requirements/local.txt
 
-#. Create a new PostgreSQL database (note: if this is the first time a database is created on your machine you might need to alter a localhost-related entry in your ``pg_hba.conf`` so as to utilize ``trust`` policy): ::
+#. Create a new PostgreSQL database using createdb_: ::
 
     $ createdb <what you've entered as the project_slug at setup stage>
 
+   .. note::
+
+       if this is the first time a database is created on your machine you might need to alter
+       a localhost-related entry in your ``pg_hba.conf`` so as to utilize ``trust`` policy, see the
+       `postgres documentation`_ for more details.
```

Comment:
I'm not 100% sure what this means tbh, I never had to set this up on my machine. If anyone has more details, I'd be more than happy to elaborate.

Reply:
I believe this was my input: in past, I did face the issue with `pg_hba.conf` localhost auth records.

Reply:
I suggest that this note stays intact: it proved itself helpful to me in past, no reason to assume the opposite for everyone else. It might require clarifications, though.

Reply:
I usually set up password for the `postgres` database user and change the authentication mode to md5:

in Ubuntu would be something like this:

```bash
# runs the "psql" command as a "postgres" linux user, and changes the password of "postgres" database user
# yes, when you install postgres on linux (at least on Ubuntu), a linux user named "postgres" is created

sudo -u postgres psql postgres -c "ALTER USER postgres WITH PASSWORD 'new_password';"
```
change `pg_hba.conf` configuration file and enable md5 authentication

>         in ubuntu, this file is located in:
>        /etc/postgresql/POSTGRESH_VERSION_NUMBER/main/pg_hba.conf
>        
>        if the version of postgres is 9.6
>
>       
>        /etc/postgresql/9.6/main/pg_hba.conf
>        
>        if is 10
>        
>        /etc/postgresql/10/main/pg_hba.conf


open and change the pg_hba.conf
from:
```
local   all             all                                     peer
```
to
```
local   all             all                                     md5
```
restart postgresql

```
sudo service postgresql restart

```

Reply:
here is a simple, and good documentation on how to configure postgresql for first use

http://suite.opengeo.org/docs/latest/dataadmin/pgGettingStarted/firstconnect.html

Reply:
I'd really like to add more details on this, but it seems to be a bit overwhelming to include in this page.

I've looked at this to see how different it is on macOS. I've re-installed `postgresql` with Homebrew and it looks like the setup is a bit different. The `pg_hba.conf` file is located under `/usr/local/var/postgres/pg_hba.conf` and it uses the `trust` policy. The DB server is running as the current user (not postgres), but has no password either.

Finally, I have no way to test how this would work under Windows...

I've deleted `docs/installing_postgres.rst` for now, I could either add it back and elaborate on this topic, or link to that external documentation.